msdm/core/problemclasses/mdp/deterministic_shortest_path.py

```diff
@@ -0,0 +1,26 @@
+from abc import abstractmethod
+from msdm.core.problemclasses.mdp import MarkovDecisionProcess
+from msdm.core.distributions import Distribution, Multinomial
+
+class DeterministicShortestPathProblem(MarkovDecisionProcess):
+    """
+    A DeterministicShortestPathProblem is a special case of an MDP, with two small differences:
```

Comment:
Should we require that all rewards are negative as well?

Reply:
done! Not sure how to programmatically enforce this, though. Realize that adding an assertion to the abstract class is a bit awkward since the abstract class needs to inspect the return value, but normally that's something the subclass handles. Going to comment, but not enforce programmatically for now, I think.